### PR TITLE
link by line node

### PR DESCRIPTION
**Pull request by stuartlynn**
Adds a node to link one table of points to another by lines based on a mutual key (category etc). Can use either a straight line or a great circle.

Need to add example code.

### CODE REVIEW

**Comment by stuartlynn** [comment]
@javisantana @rochoa This is the last node of the ones we need to the first analysis pipeline. It connects N to N points with a category by lines. You can either use straight lines or great circles.

File: lib/node/nodes/link-by-line.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+var Node = require('../node');
+var dot = require('dot');
+
+var TYPE = 'link-by-line';
+
+var PARAMS = {
+    source_points      : Node.PARAM.NODE(Node.GEOMETRY.POINT),
```
Comment (rochoa) [inline lib/node/nodes/link-by-line.js]: Do not indent keys: values. Use just one space to separate them.

File: lib/node/nodes/link-by-line.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+    dest_points        : Node.PARAM.NODE(Node.GEOMETRY.POINT),
+    id_column_source   : Node.PARAM.STRING(),
+    id_column_dest     : Node.PARAM.STRING(),
+    use_great_circle   : Node.PARAM.BOOLEAN()
```
Comment (rochoa) [inline lib/node/nodes/link-by-line.js]: Can we make this optional?

File: lib/node/nodes/link-by-line.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+var featureColumns = function(columns,prefix){
+    return columns.filter(function(col){ 
+        return ['the_geom','the_geom_webmercator','cartodb_id'].indexOf(col) === -1;
+    }).map(function(col){ 
+        return prefix+'.'+col + ' as '+prefix+'_'+col;
+    }).join(',');
+};
+
+LinkByLine.prototype.sql = function(){
+    var lineFunc      = this.use_great_circle ? 'CDB_GreatCircle' : 'ST_MakeLine';
```
Comment (rochoa) [inline lib/node/nodes/link-by-line.js]: Same here: use just one space to assign a variable.

File: lib/node/nodes/link-by-line.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+module.exports.PARAMS = PARAMS;
+
+var linkByLineTemplate = dot.template([
+    'with source as(',
+    '  select * from ({{=it.sourceQuery}}) a ' ,
+    '),',
+    'dest as (',
+    '  select * from ({{=it.destQuery}}) b ',
+    ')',
+    'select {{=it.lineFunc}}(source.the_geom, dest.the_geom) as the_geom, {{=it.sourceColumns}}, {{=it.targetColumns}}',
```
Comment (rochoa) [inline lib/node/nodes/link-by-line.js]: Should we select the `cartodb_id` from the source node?

File: lib/node/nodes/link-by-line.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+};
+
+var LinkByLine = Node.create(TYPE, PARAMS, {cache: true});
+
+module.exports = LinkByLine;
+module.exports.TYPE = TYPE;
+module.exports.PARAMS = PARAMS;
+
+var linkByLineTemplate = dot.template([
+    'with source as(',
```
Comment (rochoa) [inline lib/node/nodes/link-by-line.js]: Why don't we use a join instead of the CTEs?

**Pull request merged**